Tasks/DeployAzureResourceGroup/Utility.ps1

```diff
@@ -98,7 +99,16 @@ function Get-AzureUtility
     else
     {
         Check-AzureRMInstalled
-        $azureUtilityRequiredVersion = $azureUtilityNewVersion
+        $minimumAzureVersion = New-Object System.Version(1, 0, 3)
+        $versionCompatible = Get-AzureVersionComparison -AzureVersion $currentVersion -CompareVersion $minimumAzureVersion
+		if(!$versionCompatible)
```

Comment:
This may not scale as we go forward. Should you refactor to have a command/method to return suitable supported version based on inputs and the callers simply use switch case to set azureUtilityRequiredVersion?